snapshot-recorder/main.py

```diff
@@ -0,0 +1,180 @@
+        data_file_path: The file path to save the downloaded data.
+    """
+    with open(query_file_path) as f:
+        headcount_json = f.read()
+    query_dto = AggregationQueryExecutionDTO.from_json(headcount_json)
+
+    api_client = ApiClient(config, 'Accept', 'text/csv')
+    query_api = DataQueryApi(api_client)
+
+    logger.info(f"Executing query: {query_file_path}")  # Combined messages
+    response = query_api.aggregate_without_preload_content(query_dto)
+    if not 200 <= response.status <= 299:
+        raise Exception(f"API request failed with status code {response.status}: {response.data.decode()}")
+
+    with open(data_file_path, mode='w') as f:
+        f.write(response.data.decode())
+    logger.info(f"Data saved to temp file: {data_file_path}")  # Combined messages
+
+
+def upload_data(config: Configuration, data_file_path: str) -> None:
+    """
+    Uploads data using the Data API.
+    Args:
+        config: The API configuration object.
+        data_file_path: The file path to the data to be uploaded.
+    """
+    api_client = ApiClient(config, 'Content-Type', 'application/octet-stream')
+    upload_api = DataUploadApi(api_client)
+
+    with open(data_file_path, 'rb') as f:
+        data = f.read()
+
+    filename = os.path.basename(data_file_path)
+    logger.info(f"Uploading data: {data_file_path}")
+    upload_api.v1_data_upload_files_filename_put(filename=filename, body=data)
+
+
+def clean_temp_dir(temp_dir: str, temp_dir_created: bool, temp_data_file_path: str) -> None:
+    """Deletes the temporary file and directory (if created).
+
+    Args:
+        temp_dir: The directory where the temporary file is located.
+        temp_dir_created: A boolean indicating whether `temp_dir` was created.
+        temp_data_file_path: The full path to the temporary file.
+    """
+    if os.getenv('KEEP_TEMP_FILE', '').lower() == 'true':
+        return
+
+    os.remove(temp_data_file_path)
+    logger.info(f"Temporary file {temp_data_file_path} removed.")
+    if temp_dir_created:
+        os.rmdir(temp_dir)
+        logger.info(f"Temporary directory {temp_dir} removed.")
+
+
+def main() -> None:
+    try:
+        logger.info("Snapshot recorder started.")
+
+        args = parse_args()
+        load_dotenv(dotenv_path='.env.time-machine', override=True)
+        load_dotenv(dotenv_path='.env', override=True)
+        api_config = load_api_configuration()
+
+        temp_dir = os.getenv('TEMP_STORAGE_PATH', default='tmp_storage')
+        temp_dir_created = ensure_temp_dir_exists(temp_dir)
+        temp_data_file_path = get_temp_file_path(temp_dir, args.query_file_path)
+
+        download_data(api_config, args.query_file_path, temp_data_file_path)
+
+        upload_data(api_config, temp_data_file_path)
+
+        clean_temp_dir(temp_dir, temp_dir_created, temp_data_file_path)
+
+        logger.info("Snapshot recorder finished.")
+    except Exception as e:
+        logger.error(f"An unexpected error occurred: {e}")
```

Comment:
Clean up in the event of failure

Reply:
Decided to keep files to be able to upload it later.